tests/Fusonic/Linq/Test/MiscTest.php

```diff
@@ -36,19 +32,25 @@ public function testConcat_ReturnsConcatenatedElements()
         $this->assertEquals("d", $all[3]);
     }
 
-    public function testConcat_ThrowsArgumentExceptionIfNoTraversableArgument()
+    /**
+     * @expectedException InvalidArgumentException
```

Comment:
I (partially) agree with your first statement. However, there is no difference between the annotation and the old wrapper method. No unit test used the exception object afterwards. If I want to make sure that the correct exception is thrown during testing I would additionally use `@expectedExceptionMessage`; example from FCP project:

```php
/**
 * @covers ModuleSetup::createFromDatabaseVersion
 *
 * @expectedException \LogicException
 * @expectedExceptionMessage Invalid config file for "johndoe" module.
 */
public function testCreateFromDatabaseVersion_corruptedModuleConfigFile()
{
    // "johndoe" module does not exist.

    ModuleSetup::createFromDatabaseVersion(
        "johndoe",
        \Core::getDB()->getConnection()
    );
}
```

A lot of unit tests in `fusonic/linq` could use `@expectedExceptionMessage` to fully eliminate the possibility of incorrect asserts, but I just wanted to get rid of the `TestBase` class :wink: